packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -386,6 +387,15 @@ class NetworkController extends DisposableController
       serviceConnection.errorBadgeManager.incrementBadgeCount(NetworkScreen.id);
     }
   }
+
+  Future<void> _fetchFullDataBeforeExport() async {
+    for (final NetworkRequest item in filteredData.value) {
+      // Check if the request is of type DartIOHttpRequestData and fetch full data
+      if (item is DartIOHttpRequestData) {
+        await item.getFullRequestData();
+      }
+    }
+  }
```

Comment:
Would it make sense to await them all in one operation instead of awaiting each call to `getFullRequestData` before requesting the next? So something like:

```dart
Future<void> _fetchFullDataBeforeExport() =>
    Future.wait(filteredData.value.whereType<DartIOHttpRequestData>());
```

Reply:
Yes, this will optimise it, I had thought of doing it earlier but I missed changing it.
Thanks @julemand101 

Changed it to -

`Future<void> _fetchFullDataBeforeExport() =>

      Future.wait(
        filteredData.value
            .whereType<DartIOHttpRequestData>()
            .map((item) => item.getFullRequestData()),
      );`